electron/main.js

```diff
@@ -136,13 +161,14 @@ const showMainWindow = () => {
       preload: PRELOAD_JS,
       webviewTag: true,
     },
-    width: config.WINDOW.MAIN.DEFAULT_WIDTH,
-  });
+    width: mainWindowState.width,
+    x: mainWindowState.x,
+    y: mainWindowState.y,
+  };
 
-  const isFullScreen = settings.restore(SETTINGS_TYPE.FULL_SCREEN, false);
-  const windowBounds = settings.restore(SETTINGS_TYPE.WINDOW_BOUNDS, main.getBounds());
-  main.setBounds(windowBounds);
-  if (isFullScreen) {
+  main = new BrowserWindow(options);
+  mainWindowState.manage(main);
+  if (typeof mainWindowState.isMaximized === 'undefined' && isFullScreen === true) {
```

Comment:
Why undefined? This should be true or false. Also: Does this account for the difference between macOS and Windows?

Reply:
This is part of the upgrade process to `electron-window-state`. This line comes only to use if there is no `isMaximized` value (which comes from electron-window-state) and if we still have the old setting `isFullScreen` set to `true`. In the future the can remove this line as `electron-window-state` will maximize the window for us if appropriate.

Reply:
Let's write the code in a way that this is clear to everyone then. The line above does not hint at being related to the upgrade process at all. Pull it into a descriptive const and add a comment.